.eslintrc.json

```diff
@@ -0,0 +1,121 @@
+{
+    "root": true,
+    "ignorePatterns": ["projects/**/*"],
+    "overrides": [
+        {
+            "files": ["*.ts"],
+            "parserOptions": {
+                "project": ["tsconfig.json", "e2e/tsconfig.json"],
+                "createDefaultProgram": true
+            },
+            "extends": [
+                "plugin:@angular-eslint/ng-cli-compat",
+                "plugin:@angular-eslint/ng-cli-compat--formatting-add-on",
+                "plugin:@angular-eslint/template/process-inline-templates"
+            ],
+            "rules": {
+                "@angular-eslint/component-selector": [
+                    "error",
+                    {
+                        "type": "element",
+                        "prefix": "wave",
+                        "style": "kebab-case"
+                    }
+                ],
+                "@angular-eslint/directive-selector": [
+                    "error",
+                    {
+                        "type": "attribute",
+                        "prefix": "wave",
+                        "style": "camelCase"
+                    }
+                ],
+                "@typescript-eslint/consistent-type-definitions": "error",
+                "@typescript-eslint/dot-notation": "off",
+                "@typescript-eslint/explicit-member-accessibility": [
+                    "off",
+                    {
+                        "accessibility": "explicit"
+                    }
+                ],
+                "@typescript-eslint/member-ordering": [
+                    "error",
+                    {
+                        "default": {
+                            "memberTypes": [
+                                "public-field",
+                                "protected-field",
+                                "private-field",
+                                "constructor",
+                                "public-method",
+                                "protected-method",
+                                "private-method"
+                            ]
+                        }
+                    }
+                ],
+                "camelcase": "off",
```

Comment:
why turn camelCase off and then enable it again for individual features? Why don't we use default naming-conventions?

Reply:
Well, I guess we want to use constants like GENERIC_FIELDS and we can't do this with camelCase. Furthermore, our types from the backend often use snake_case and for enums we usually use PascalCase. So camelCase for everything doesn't work out. But if you have alternative suggestions, feel free to suggest some.